src/erl_optics.erl

```diff
@@ -315,8 +333,24 @@ create_optics(Name) ->
     end.
 
 get_lens(Key) ->
-    foil:lookup(?NS, Key).
-
+    case foil:lookup(?NS, Key) of
+        {ok, Ptr} ->
+            {ok, Ptr};
+        Err ->
+            {error, {Err,Key}}
+    end.
 
 get_optics() ->
     foil:lookup(?NS, optics).
+
+get_lenses(Module) ->
+    case erlang:function_exported(Module, get_optics_lenses, 0) of
+        true ->
+            {ok, Lenses} = Module:get_optics_lenses(),
```

Comment:
It looks like this could cause a `badmatch`. Do you think it should be wrapped in a try/catch?